symengine/sets.h

```diff
@@ -160,11 +165,28 @@ class Interval : public Set
     virtual RCP<const Set> set_intersection(const RCP<const Set> &o) const;
     virtual RCP<const Boolean> contains(const RCP<const Basic> &a) const;
     virtual vec_basic get_args() const;
+
+    inline const RCP<const Number> &get_start() const
+    {
+        return start_;
+    }
+    inline const RCP<const Number> &get_end() const
+    {
+        return end_;
+    }
+    inline const bool &get_left_open() const
```

Comment:
Right, simple types like `int`, `double`, `bool` (also I think `std::complex`) should be returned by value.